src/legacy/core_plugins/data/public/search/search_bar/components/search_bar.tsx

```diff
@@ -70,10 +70,14 @@ export interface SearchBarProps {
   isRefreshPaused?: boolean;
   refreshInterval?: number;
   showAutoRefreshOnly?: boolean;
-  showSaveQuery?: boolean;
   onRefreshChange?: (options: { isPaused: boolean; refreshInterval: number }) => void;
+  // Show when user has privileges to save
+  showSaveQuery?: boolean;
+  // User has saved the current state as a saved query
   onSaved?: (savedQuery: SavedQuery) => void;
+  // User has modified the saved query, your app should persist the update
   onSavedQueryUpdated?: (savedQuery: SavedQuery) => void;
+  // User has cleared the active query, your app should clear the entire query bar
```

Comment:
👍 I'm a fan of these comments.

Reply:
I'm a fan of deleting these - coming soon :+1:

Reply:
@lizozom please let's discuss 'deleting these arguments' and what entails for saved queries.